zookeeper-api/src/main/java/org/apache/helix/zookeeper/impl/client/SharedZkClient.java

```diff
@@ -402,6 +478,13 @@ public Stat writeDataGetStat(String path, Object datat, int expectedVersion) {
     return _innerSharedZkClient.writeDataReturnStat(path, datat, expectedVersion);
   }
 
+  @Override
+  public void asyncCreate(String path, Object datat, CreateMode mode, long ttl,
+      ZkAsyncCallbacks.CreateCallbackHandler cb) {
+    checkIfPathContainsShardingKey(path);
+    _innerSharedZkClient.asyncCreate(path, datat, mode, ttl, cb);
+  }
+
   @Override
   public void asyncCreate(String path, Object datat, CreateMode mode,
```

Comment:
Can we use ZkClient.TTL_NOT_SET to simplify this implementation?

Reply:
Yes, made the change.